remote-files: expand and improve remote Ignition file section (#473)
coreos/fedora-coreos-docs: modules/ROOT/pages/remote-ign.adoc

With Ignition, you are not limited to the configuration provided locally to a system and you can retrieve other Ignition configs from a remote source. Those configs will then either replace or be merged into the existing config.

The complete list of supported protocols and related options for remote Ignition files is described in the https://coreos.github.io/ignition/specs/[Ignition specification].

The following examples show how to retrieve an Ignition file from a remote source. They are both set to replace the current configuration with a remote Ignition file.

.Retrieving a remote Ignition file via HTTPS
[source,yaml]
----
variant: fcos
version: 1.4.0
ignition:
  config:
    replace:
      source: https://example.com/sample.ign
      verification:

>         hash: sha512-e2bb19fdbc3604f511b13d66f4c675f011a63dd967b97e2fe4f5d50bf6cb224e902182221ba0f9dd87c0bb4abcbd2ab428eb7965aa7f177eb5630e7a1793e2e6

Reviewer: In practice, `verification` isn't used much with HTTPS.  I'd be inclined to use it only in the HTTP example.

Author: Removed `verification` from all but the plain HTTP example.

      ssh_authorized_keys:
        - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDHn2eh...
----

Retrieving remote Ignition files via plain HTTP is also possible as shown below.

> WARNING: Retrieving a remote Ignition file via HTTP is discouraged for obvious reasons. When using HTTP, it is advisable to use the verification option to ensure the contents haven't been tampered with.

Reviewer: Obvious or not, let's spell it out:

```suggestion
WARNING: Retrieving a remote Ignition config via HTTP exposes the contents of the config to anyone monitoring network traffic. When using HTTP, it is advisable to use the verification option to ensure the contents haven't been tampered with.
```

Author: Applied your wording fixes.

.Retrieving a remote Ignition file via HTTP
[source,yaml]
----
variant: fcos
version: 1.4.0
ignition:
  config:
    replace:
      source: http://example.com/sample.ign
      verification:
        hash: sha512-e2bb19fdbc3604f511b13d66f4c675f011a63dd967b97e2fe4f5d50bf6cb224e902182221ba0f9dd87c0bb4abcbd2ab428eb7965aa7f177eb5630e7a1793e2e6
----

> In case you need to retrieve a remote Ignition file but have no direct access to the remote host you can specify a proxy for both plain HTTP and HTTPS. Hosts that should be excluded from proxying can also be specified.

Reviewer: ```suggestion
If you need to retrieve a remote Ignition file but have no direct access to the remote host, you can specify a proxy for plain HTTP and/or HTTPS. You can also specify hosts that should be excluded from proxying.
```

Author: Applied your wording fixes.
